lib/indocker/repositories/clonner.rb

```diff
@@ -9,9 +9,41 @@ def initialize(configuration, logger)
   def clone(session, repository)
     raise ArgumenError.new("only git repositories should be clonned") if !repository.is_git?
 
-    session.exec!("rm -rf #{repository.clone_path} && mkdir -p #{repository.clone_path}")
-    
-    git_command = "git clone -b #{repository.branch} --depth 1 #{repository.remote_url} #{repository.clone_path}"
+    repository_remote_url = session.exec!(
+      build_git_remote_url_command(
+        path: repository.clone_path
+      )
+    ).stdout_data.chomp
+
+    @logger.debug("server remote_url:  #{repository_remote_url.inspect}, indocker remote_url: #{repository.remote_url.inspect}")
+
+    git_command = if repository_remote_url == repository.remote_url
+      build_force_pull_command(
+        target_path: repository.clone_path,
+        branch_name: repository.branch,
+        remote_url:  repository.remote_url,
+      )
+    else
+      build_clone_command(
+        target_path: repository.clone_path,
+        branch_name: repository.branch,
+      )
+    end
+
     session.exec!("ssh-agent bash -c 'ssh-add ~/.ssh/#{repository.ssh_key}; #{git_command}'")
   end
-end
+
+  private
+
+    def build_git_remote_url_command(path:)
+      "mkdir -p #{path} && cd #{path} && git config --get remote.origin.url"
```

Comment:
I think it will be simpler to read these commands if they will be based on Array with join, e.g.
[
  "mkdir -p #{path}",
  "cd #{path}",
  ...
].join(" && ")

Reply:
👍